microsoft-edge/devtools-guide-chromium/json-viewer/json-viewer.md

```diff
@@ -0,0 +1,67 @@
+---
+title: View formatted JSON
+description: How to use the JSON viewer to view formatted and highlighted JSON responses in a browser tab.
+author: MSEdgeTeam
+ms.author: msedgedevrel
+ms.topic: conceptual
+ms.prod: microsoft-edge
+ms.date: 01/04/2023
+---
+# View formatted JSON
+
+Use the **JSON viewer** to automatically format and highlight JSON responses and files in browser tabs.
+
+The **JSON viewer** changes the returned JSON data to make it easier to read. Sometimes a web server responds to HTTP requests by returning data encoded as JSON.  JSON data can be difficult to read when it's formatted as a single long, concatenated line of text. The same can occur when opening a JSON file from disk.
+
+The **JSON viewer** improves readability of JSON data in several ways:
+
+*  The JSON syntax is highlighted with different colors.
+*  Object properties are displayed on their own lines and indented.
+*  Objects can be collapsed or expanded.
+
+The **JSON viewer** tool is included in Microsoft Edge starting with version 110.
+
+
+<!-- ====================================================================== -->
+## View reformatted JSON server responses
+
+To view a JSON response from a web server as reformatted JSON:
+
+1. Open a new tab or window in Microsoft Edge.
+
+1. Type a URL that returns JSON data in the address bar. For example, use [this sample JSON response](https://microsoftedge.github.io/Demos/json-dummy-data/256KB-min.json). You can find more samples JSON responses at [MicrosoftEdge / Demos > json-dummy-data](https://github.com/MicrosoftEdge/Demos/tree/main/json-dummy-data).
+
+1. Microsoft Edge detects that the returned data is JSON and formats it automatically:
+
+   ![JSON data returned by a web server, formatted and syntax-highlighted by the JSON viewer](./images/json-response.png)
+
+
+<!-- ====================================================================== -->
+## View reformatted JSON files
+
+To view a JSON file stored on your device as reformatted JSON:
+
+1. Open a new tab or window in Microsoft Edge.
+
+1. Press `Ctrl`+`O` on Windows and Linux, or `Cmd`+`O` on macOS and choose the JSON file you want to open.
+
+1. Microsoft Edge detects that the file contains JSON and formats it automatically:
+
+   ![JSON data stored in a local file, formatted and syntax-highlighted by the JSON viewer](./images/json-file.png)
+
+
+<!-- ====================================================================== -->
+## Detect invalid JSON data
+
+JSON data can sometimes be invalid. For example, the following JSON is invalid:
+
+*  Omitting double quotes around a key name.
+*  Having a trailing comma after the last key/value pair.
```

Comment:
```suggestion
*  Having a trailing comma after the last key/value pair.
*  Missing colon.
* Unterminated.
*  Binary data.
```
tech review / clarify.  
sync w/ https://github.com/MicrosoftEdge/Demos/tree/main/json-dummy-data#invalid-json

Reply:
Hm, I think this is making the article unnecessarily complex. We're not documenting the JSON format here, nor do we really need to list all of the ways it can be invalid. We're documenting the tool only. I would personally stick with the 2 examples of invalid JSON we already have.

Reply:
And, it's ok for me if the Demos repo has more invalid sample JSON files than the ones we list here. The Demos repo isn't a perfect 1:1 mapping of what we have in the docs, and doesn't need to be. It can serve other purposes. In this case, the invalid JSON files on the repo are useful to test how the tool behaves in a variety of cases (such as binary data).

Reply:
this thread can be Resolved